pandas/io/common.py

```diff
@@ -194,19 +211,26 @@ def get_filepath_or_buffer(
         req.close()
         return reader, encoding, compression, True
 
-    if is_s3_url(filepath_or_buffer):
-        from pandas.io import s3
+    if is_fsspec_url(filepath_or_buffer):
+        import fsspec
+        scheme = parse_url(filepath_or_buffer).scheme
+        filesystem = fsspec.filesystem(scheme)
+        file_obj = filesystem.open(filepath_or_buffer, mode=mode or "rb")
```

Comment:
Do we need to pass through the encoding? to .open() You will potentially also fix this if you do so: https://github.com/pandas-dev/pandas/issues/26124 :)